blockprotocol/blockprotocol
# Add client-side Sentry React instrumentation to Wordpress Plugin

## 🌟 What is the purpose of this PR?

**Context:** we're trying to debug a number of user errors that are proving elusive.

This PR adds client-side React instrumentation for React to the admin view of post editing. The plugin aims to be a good citizen on the installation by not overwriting existing configurations if there are any.

The Sentry configuration includes console error/warn/assert logging so we can track bugs. The reporting only happens for JS files matching the `plugins/blockprotocol` regex.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1204000740778935/1204197991701675/f) _(internal)_

## 🔍 What does this change?

- Changes how Sentry is configured to extract common settings for client/server-side
- Conditionally add Sentry reporting on the admin editing view
- Add new webpack entry for `sentry`
- Add new DSN for client side sentry along with its config

## ⚠️ Known issues

- We are unable to capture errors if other plugins/scripts take over the sentry instance - differently from the WP plugin where we manually instruments some errors.

## 🐾 Next steps

Release after #1181 is merged

## ❓ How to test this?

1.  Checkout the branch, start WP instance with newly built JS. 
1.  Configure BP plugin with an API key
1.  Go to a post to edit it, issue a `console.error("Boom!")` in the console of the post editor page
1.  Confirm that event was tracked in sentry.

You can also trigger an error from react by adding 
```tsx
      <button onClick={() => methodDoesNotExist()}>Break the world</button>;
```
to the `edit.tsx` return and click the button after creating a block. Sentry should also pick this up.

## 📹 Demo

<details><summary>Running the above test steps</summary>


![image](https://user-images.githubusercontent.com/6988668/225721949-99c6b9aa-33a6-4d54-811b-b55225c773b6.png)

![image](https://user-images.githubusercontent.com/6988668/225722403-8c171e33-1672-4301-aaa2-62a77446c401.png)


</details>

<details><summary>Screenshots of how Sentry behaves with enabling/disabling telemetry</summary>


![image](https://user-images.githubusercontent.com/6988668/225722130-8d89dc41-c9be-48cf-9f87-7dbae7278b56.png)

![image](https://user-images.githubusercontent.com/6988668/225722142-85e3b1cf-22f2-4e34-834e-24777ac43809.png)


</details>

I've also tested the plugin with another JS-side WP Sentry extension. Our config is simply overwritten. We do try not to create a sentry instance if we can see there's already one.
